File: go/1402_Reducing_Dishes.go
import (
	"sort"
)

func maxSatisfaction(satisfaction []int) int {
    sort.Sort(sort.Reverse(sort.IntSlice(satisfaction)))
    var tempSum, totalSum = 0, 0
    for _, dish := range satisfaction  {
        tempSum += dish
        if (tempSum < 0) {
            break;
        }
        totalSum += tempSum
    }

    return totalSum
}

File: php/1400/1402_Reducing_Dishes.php
<?php

class Solution 
{

    /**
     * @param Integer[] $satisfaction
     * @return Integer
     */
    function maxSatisfaction($satisfaction) 
    {
        rsort($satisfaction);
        $satisfactionCount = count($satisfaction);
        $tempSum = 0;
        $totalSum = 0;
        foreach ($satisfaction as $dish) {
            $tempSum += $dish;
            if ($tempSum < 0) {
                break;
            }
            $totalSum += $tempSum;
        }

        return $totalSum;
    }
}

